onnx/shape_inference/implementation.cc

```diff
@@ -486,21 +486,43 @@ class ShapeInferenceImplBase {
     }
   }
 
-  void process(const NodeProto& n, std::unordered_map<std::string, const AttributeProto*> attr_map) {
-    NodeProto copy_n(n);
-    // Add attribute information into the temporary node
-    copy_n.clear_attribute();
-    for (const auto& attr : n.attribute()) {
-      if (attr.has_ref_attr_name()) {
-        if (attr_map.count(attr.ref_attr_name())) {
-          auto copy_attr = *attr_map[attr.ref_attr_name()];
-          copy_attr.set_name(attr.name());
-          copy_n.add_attribute()->CopyFrom(copy_attr);
+  void replaceAttrParams(NodeProto& n, std::unordered_map<std::string, const AttributeProto*> attr_map) {
```

Comment:
Why is this called "replaceAttrParams"?
What does "Param" stand for? Is it an attribute in NodeProto?

Reply:
Also confused about Params here. How about calling `replaceAttrValues`?

Reply:
An ONNX function has parameters of two kinds (just like an ONNX ops): the regular parameters (tensors, etc., whose values will be known only at runtime) and attribute-parameters ... "Param" here is shorthand for "Parameter". As discussed elsewhere, I changed the name to "AttrRefs", though I think both names should be fine.